### ISSUE TITLE
[BUG] decimal.DivisionUndefined error in arbitrage strategy on Windows

### ISSUE THREAD

**Issue by bedud5495** [CoinAlpha/hummingbot#1358]
**Describe the bug**
2020-01-29 00:21:08,001 - 10 - hummingbot.core.clock - ERROR - Unexpected error running clock tick.
Traceback (most recent call last):
  File "clock.pyx", line 119, in hummingbot.core.clock.Clock.run_til
  File "arbitrage.pyx", line 172, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_tick
  File "arbitrage.pyx", line 344, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair
  File "arbitrage.pyx", line 364, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair_inner
  File "arbitrage.pyx", line 469, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_find_best_profitable_amount
decimal.InvalidOperation: [<class 'decimal.DivisionUndefined'>]

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters: ...
2. Start the bot using this command: ...
...

**Screenshots**
// If applicable, add screenshots to help explain your problem.

**Release version**
// Include your bot's version number (Can be found at the upper left corner of your CLI).

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**
**Optional: your ETH address:** (If you want to participate in the bug bounty)

**Comment by Hashira-Lars** [comment]
Hi @bedud5495 could you upload your config file and logs so we can further investigate

**Comment by bedud5495** [comment]
#####################################
###   Arbitrage strategy config   ###
#####################################

template_version: 2

# The following configuations are only required for the
# arbitrage strategy

# Exchange and token parameters
primary_market: binance
secondary_market: bittrex
primary_market_trading_pair: BTC-USDT
secondary_market_trading_pair: BTC-USDT

# Minimum profitability target required to place an order
# Expressed in decimals: 0.01 = 1% target profit
min_profitability: 0.001

# For more detailed information, see:
# https://docs.hummingbot.io/strategies/arbitrage/#configuration-parameters

**Comment by bedud5495** [comment]
[logs_conf_arbitrage_strategy_1.log](https://github.com/CoinAlpha/hummingbot/files/4136039/logs_conf_arbitrage_strategy_1.log)

![010](https://user-images.githubusercontent.com/18395410/73486667-e2165280-43a5-11ea-8bbf-b5c2f8172f67.png)

**Comment by Hashira-Lars** [comment]
Hi @bedud5495 

(Arbitrage/Binance-Bittrex/BTC-USDT/Paper-trade[on]/Telegeram[on])

- Tested with windows binary for 3hours cannot reproduce error
- Tested with docker windows local for an hour cannot reproduce behavior

1. Base on your configs your using the latest version master as bittrex is supported in paper-trade mode (also possible using dev)
2. Base on your logs your stopping the bot (unknown what your doing) then re-starting the bot
3. Base on your logs telegram is enabled

Question's : 

Could you provide the step's you've taken.
Did the error happened intermittently or not

Suggestion : 

- Could you indicate what your discord username so we can follow up things faster in discord
- Can you try to re-create new instance (if docker) and re-create new strategy file then re-test (let us know how it goes if the error still occurs)

**Comment by Hashira-Lars** [comment]
After 2hours have reproduce the behavior with docker windows local.

one of the error's indicated : 

```2020-01-31 02:28:42,030 - 10 - hummingbot.strategy.arbitrage.arbitrage - INFO - Market order completed on binance: sell://BTCUSDT/c4014d0ff388b90670daf0fdb8 [clock=2020-01-31 02:28:41+00:00]
2020-01-31 02:28:42,046 - 10 - hummingbot.core.network_iterator - ERROR - Unexpected error while processing event 107.
Traceback (most recent call last):
  File "pubsub.pyx", line 165, in hummingbot.core.pubsub.PubSub.c_trigger_event
  File "paper_trade_market.pyx", line 167, in hummingbot.market.paper_trade.paper_trade_market.OrderBookMarketOrderFillListener.c_call
  File "composite_order_book.pyx", line 53, in hummingbot.core.data_type.composite_order_book.CompositeOrderBook.record_filled_order
TypeError: unsupported operand type(s) for +=: 'decimal.Decimal' and 'float'
```

Will try to update windows local time settings if it will fix the issue (will update later on)

Logs : 
[logs_conf_arbitrage_strategy_0.log](https://github.com/CoinAlpha/hummingbot/files/4137470/logs_conf_arbitrage_strategy_0.log)

**Comment by pacovi** [comment]
I get the same issues with a different pair and not with Bittrex. My config below, I'm using docker in a Google Computer Engine.
```
#####################################
###   Arbitrage strategy config   ###
#####################################
template_version: 3
strategy: arbitrage
# The following configuations are only required for the
# arbitrage strategy
# Exchange and token parameters
primary_market: coinbase_pro
secondary_market: binance
primary_market_trading_pair: BTC-EUR
secondary_market_trading_pair: BTC-EUR
# Minimum profitability target required to place an order
# Expressed in percentage value, e.g. 1 = 1% target profit
min_profitability: 0.3
# For more detailed information, see:
# https://docs.hummingbot.io/strategies/arbitrage/#configuration-parameters
```

Release version 0.26.1

To reproduce just run the bot and i'll show up the behaviour. Also the error does show in the interface itself and also on the log file.
LOGFILE
```
decimal.InvalidOperation: [<class 'decimal.DivisionUndefined'>]
2020-05-16 10:55:13,001 - 10 - hummingbot.core.clock - ERROR - Unexpected error running clock tick.
Traceback (most recent call last):
  File "clock.pyx", line 119, in hummingbot.core.clock.Clock.run_til
  File "arbitrage.pyx", line 172, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_tick
  File "arbitrage.pyx", line 344, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair
  File "arbitrage.pyx", line 364, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair_inner
  File "arbitrage.pyx", line 469, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_find_best_profitable_amount
decimal.InvalidOperation: [<class 'decimal.DivisionUndefined'>]
2020-05-16 10:55:14,002 - 10 - hummingbot.core.clock - ERROR - Unexpected error running clock tick.
Traceback (most recent call last):
  File "clock.pyx", line 119, in hummingbot.core.clock.Clock.run_til
  File "arbitrage.pyx", line 172, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_tick
  File "arbitrage.pyx", line 344, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair
  File "arbitrage.pyx", line 364, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_process_market_pair_inner
  File "arbitrage.pyx", line 469, in hummingbot.strategy.arbitrage.arbitrage.ArbitrageStrategy.c_find_best_profitable_amount
decimal.InvalidOperation: [<class 'decimal.DivisionUndefined'>]
```

In the interface it shows only a cryptic error:
![image](https://user-images.githubusercontent.com/6522045/82118101-3a1a6e80-9775-11ea-80dc-8867676b1f4c.png)

**Comment by pacovi** [comment]
Some update on this, still same config, able to reproduce right after the first trade tries to start, some time ago was able to do trades so not sure what was the trigger to get into the lock situation, but has since then remained. At the moment arbitrage does not work with this config. Papertrades work just fine.

I've tried the following changes to the config file and the system (GCE), all of them unsuccessful:
- Start/stop hummingbot
- restart VM
- Changing order of exchanges
- Disabling / enabling Telegram
- Changing API key in Binance
- Issuing spot order on Binance to see if it "clears" something up (some kind of queue?)
- Try other currency-pairs (e.g. USD-BTC), not working
- re-install hummingbot from scratch (new docker instance), using "latest", same results
- re-install hummingbot from scratch (new docker instance), using "development", same results

Since **it looks like a general failure with arbitrage in Binance**, should I open a new bug for this pointing at this one as "might be related"?

Will gladly provide more info if needed @mifeng

**Comment by rennel-tabing** [comment]
@pacovi I tested your config on a Windows binary installation but I am unable to reproduce the error that you are getting,
![image](https://user-images.githubusercontent.com/53868268/82250640-d4b3b280-9919-11ea-9e76-bf1ff50f78fc.png)

**Comment by pacovi** [comment]
@rennel-tabing  I'm able to reproduce consistently using a Docker-based installation on Ubuntu in GCE (cloud).
I have also re-tried with 0.27.0 and still persists.

I will try with a non-docker version, but can only do so on Linux-based machines in Cloud, unfortunately I do not have a Windows10 installation available.

**Comment by pacovi** [comment]
Hi @rennel-tabing , I am looking closely to your screenshot. With the conditions that you show it does not reproduce for me either.

As far as I can see, your balance does not trigger an "attempt" to do sell/buy orders, even though you are setting correct profitability levels and the market shows that it could be profitable (status output vs min profitability config value).

*The problem occurs when you also have enough balance to do the profitable trade*, hummingbot will try to do the trade but most often than not will not be able to issue the orders due to the clocking issue.

**Comment by vic-en** [comment]
Issue resolved in PR #1832.
